### DISCUSSION TITLE
How to set the page title when using suspense queries

### DISCUSSION THREAD

**Discussion by jasonhilldm** [vikejs/vike#1902] [Discussion]
I am using preact with TanStack query, telefunc and prisma to retrieve data from a database via the useSuspenseQuery hook. I started from boilerplate code in the vike-preact sample which uses the onRenderClient and onRenderHtml hooks, so the title is set client side via a call to document.title, using the value from pageContext.documentProps.title.

In my case, the data returned from useSuspenseQuery is used to set the value in pageContext.documentProps.title. This is not working when browsing the page normally as the browser title is not being set. But when I set a breakpoint in the onRenderClient hook and inspect the pageContext value in the browser, I can see the title value is set correctly and then the browser title then updates correctly. When I remove the breakpoint, the title does not get set correctly, so maybe there is a timing issue.

Is this due to the async nature of the suspense rendering? What is the correct way to set the title in this scenario?

**Comment by brillout** [comment]
You can have a look at how `vike-{react,vue,solid}` does it.

(Btw. having a `vike-preact` would be great, I can see the Preact community liking it a lot. Happy to give away the npm package.)

**Reply by jasonhilldm** [reply]
OK...I think it's easier to just ditch preact and revert to using react. It all works fine when using vike-react and you guys are actively maintaining that module so it makes heaps more sense.
